### ISSUE TITLE
Deserializing a template isn't possible in some instances.

### ISSUE THREAD

**Issue by jkelleyrtp** [DioxusLabs/dioxus#679]
**Problem**

It's not possible to deserialize a template in place if it contains certain characters that require some processing. We should use a `cow<'a, str>`, I think.

https://github.com/serde-rs/serde/issues/1413

**Steps To Reproduce**

Hot reload with some SVG causes the app to crash.

**Expected behavior**

The app doesn't crash with hot-reload enabled.

**Screenshots**

<img width="796" alt="Screenshot 2022-12-25 at 10 25 22 PM" src="https://user-images.githubusercontent.com/10237910/209495995-9ae12960-cc33-49da-93d5-1fd2fb64265a.png">

**Environment:**
 - Dioxus version: [master`]
 - App platform: [`web`]

**Questionnaire**

- [ ] I'm interested in fixing this myself but don't know where to start
- [ ] I would like to fix and I have a solution
- [ ] I don't have time to fix this right now, but maybe later

**Comment by jkelleyrtp** [comment]
This seems to be able to be fixed by using deserialize on a serde::Value to force the escaping into owned data first.

```rust
                let val = serde_json::from_str::<serde_json::Value>(&string).unwrap();
                // leak the value
                let val: &'static serde_json::Value = Box::leak(Box::new(val));
                let template: Template<'_> = Template::deserialize(val).unwrap();
                tx.unbounded_send(template).unwrap();
```

You could either chock it up to an issue with the design of `Template` or just accept that we need to let serde do its escape sequence thing first and then leak the resulting value.

**Comment by jkelleyrtp** [comment]
Fixed in #671
